p2p/src/protocol.rs

```diff
@@ -288,10 +288,24 @@ impl MessageHandler for Protocol {
 				self.adapter
 					.txhashset_download_update(download_start_time, 0, sm_arch.bytes);
 
-				let mut tmp = env::temp_dir();
-				tmp.push("txhashset.zip");
+				let mut tmp = self.adapter.get_tmp_dir();
```

Comment:
Perhaps now that logic would fit better in the chain, given that it already holds the preparation of the directory location?

Reply:
What do you mean?  moving `save_txhashset_to_file()` function into `chain`?

Reply:
@ignopeverell  remind ^ above question.

Reply:
I meant that whole code section, with getting the tmp dir, checking if it exists, creating it if not, appending file name, etc.

Reply:
done. moved the logic to chain